utils/test_matrix/test_matrix_test.go

```diff
@@ -25,30 +26,51 @@ func getAllFlowPackages() []string {
 	}
 }
 
+// TestListTargetPackages tests that the target packages are included in the target packages and seen packages.
 func TestListTargetPackages(t *testing.T) {
 	targetPackages, seenPackages := listTargetPackages([]string{"abc", "ghi"}, getAllFlowPackages())
 	require.Equal(t, 2, len(targetPackages))
-	require.Equal(t, 4, len(seenPackages))
 
-	// there should be 3 packages that start with "abc"
-	require.Equal(t, 3, len(targetPackages["abc"]))
+	// there should be 4 packages that start with "abc"
+	require.Equal(t, 4, len(targetPackages["abc"]))
 	require.Contains(t, targetPackages["abc"], flowPackagePrefix+"abc")
+	require.Contains(t, targetPackages["abc"], flowPackagePrefix+"abc/123")
 	require.Contains(t, targetPackages["abc"], flowPackagePrefix+"abc/def")
 	require.Contains(t, targetPackages["abc"], flowPackagePrefix+"abc/def/ghi")
 
 	// there should be 1 package that starts with "ghi"
 	require.Equal(t, 1, len(targetPackages["ghi"]))
 	require.Contains(t, targetPackages["ghi"], flowPackagePrefix+"ghi")
 
+	// there should be 5 packages that start with "abc"
+	require.Equal(t, 5, len(seenPackages))
 	require.Contains(t, seenPackages, flowPackagePrefix+"abc")
+	require.Contains(t, seenPackages, flowPackagePrefix+"abc/123")
 	require.Contains(t, seenPackages, flowPackagePrefix+"abc/def")
 	require.Contains(t, seenPackages, flowPackagePrefix+"abc/def/ghi")
 	require.Contains(t, seenPackages, flowPackagePrefix+"ghi")
 }
 
+// TestListTargetSubPackages tests that if a subpackage is specified as a target package, then the sub package and
+// all children of the sub package are also included in the target packages.
+func TestListTargetSubPackages(t *testing.T) {
+	targetPackages, seenPackages := listTargetPackages([]string{"abc/def"}, getAllFlowPackages())
+	require.Equal(t, 1, len(targetPackages))
+
+	// there should be 2 packages that starts with "abc/def"
+	require.Equal(t, 2, len(targetPackages["abc/def"]))
+	require.Contains(t, targetPackages["abc/def"], flowPackagePrefix+"abc/def")
```

Comment:
should we also check the second package?
```suggestion
	require.Contains(t, targetPackages["abc/def"], flowPackagePrefix+"abc/def")
	require.Contains(t, targetPackages["abc/def/ghi"], flowPackagePrefix+"abc/def/ghi")
```

Reply:
Thanks for the catch! I updated it slightly differently from the suggestion.

789c662d001f50a1d7661a10f7514528432bffa9